### ISSUE TITLE
[Release-1.0] Implement Redux

### ISSUE THREAD

**Issue by adilanchian** [PixelogicDev/Gruvee#24]
It looks like we have held off long enough and makes more sense to start utilizing Redux, even for our mock data. The implementation flow will go as follows:

1. Setup Members Side Menu (#17) This is currently in progress. Will implement the first Redux stuff here
2. For actions like adding/removing/editing comments/songs/playlists we will want to refactor those to also use redux
3. Depending on whats left, we might want to fully refactor to Redux or just keep current state management

**Comment by adilanchian** [comment]
Started with REDUX, but need to log actions or setup react redux devtools

**Comment by adilanchian** [comment]
Setup a middleware to utilize a logger for redux actions and such. Also setup the PlaylistView to ingest redux.

Next step is to have the component use the new redux state.

**Comment by adilanchian** [comment]
Implemented actions for AddPlaylist and DeletePlaylist. Will need to continue on with cleaning that up and implementing Redux actions!

**Comment by adilanchian** [comment]
We have implemented the songs actions and will continue working through the cleanup for the initial songs work. Essentially removing the old way of getting the songs.

**Comment by adilanchian** [comment]
Added new action for deleting song and started to implement thunks. Still in the process of setting up that hierarchy, but will finish next stream.

We also need to make sure to ignore functions in the logger so it doesn't print them out!

**Comment by adilanchian** [comment]
I have went in and been able to update our redux state. It's a much flatter way of doing it and I believe will help in the long run. Next step is to start wiring up the proper actions with this new state.

**Comment by adilanchian** [comment]
Left off at setting up deletePlaylistAction but has some errors and no logic. Will do tomorrow

**Comment by adilanchian** [comment]
Added in the add playlist action with our new flat redux state! Will continue with song actions tomorrow

**Comment by adilanchian** [comment]
**TL;DR - WE SPREAD ALL THE THINGS.**

Today we implemented the RemovePlaylist action alongside the AddNewSong action. Tweaked some class id generation in the Song and Playlist classes.

Tomorrow we will move onto the DeleteSong action as well as setting up the Comments state and start with those actions.

**Comment by adilanchian** [comment]
Added the delete song action for our redux state

**Comment by adilanchian** [comment]
### What We Did
- Added in Reselect for fancy selectors
- Fixed a bug where deleting a shared song would remove it entirely from our SongState. Ended up checking playlists to make sure that a shared song was not deleted from our state
- Started adding the AddComment action to redux. Currently need to just update the playlist with the updated song

### What We Will Do Tomorrow
- Finalize AddComment redux action
- Add in the DeleteComment redux action
- Need to alter Playlist & Song Delete to remove it's children (ie: If you delete a playlist all songs should be removed from state, and any corresponding comments should also be removed from state.

**Comment by adilanchian** [comment]
### What We Did
- Added in AddComment Redux action 
- Add in DeleteComment Redux action 
- Added in some logic for removing comments per song, BUT found out that our data structure is all jacked up for comments. It actually persists comments for songs that are shared across playlists.
- Did some slight property renaming for our actions

### What We Will Do Tomorrow

- [ ] Alter the data structure to properly handle comments for songs. They need to be a part of the playlist in order for us to keep them unique.
- [ ] Add a User state into redux 
- [ ] Setup some edit UI for Playlists/Songs/Comments

**Comment by adilanchian** [comment]
### What We Did
- Updated our data model to associated comments by playlist and their respected songs
- Updated the AddSong and AddComment actions to account for this change
- Updated the Songs Selector to account for this change
- HAD OUR FIRST STANDUP

### What We Will Do Tomorrow
- [ ] Finish up updating the DeletePlaylists, DeleteSong and DeleteComment Redux actions to account for new data model changes
- [ ] Add User state to Redux
- [ ] Setup some edit UI for Playlists/Songs/Comments (BONUS ITEM)

**Comment by adilanchian** [comment]
### What We Did
- Tweaked delete comments when removing a song
- Tweaked delete comment redux action

### What We Will Do Tomorrow
- [ ] Fixed circular dependency issue
- [ ] Add User state to Redux
- [ ] Merge this branch into `release-1.0` and start a new issue!

**Comment by adilanchian** [comment]
IN PR. O M G.

**Comment by adilanchian** [comment]
bye bye.
